soc/riscv/riscv-privilege/andes_v5/l2_cache.c

```diff
@@ -88,6 +91,7 @@ static void andes_v5_l2c_enable(void)
 
 static int andes_v5_l2c_init(const struct device *dev)
 {
+#if DT_NODE_HAS_STATUS(DT_INST(0, syscon), okay)
```

Comment:
Isn't `device_is_ready` failing when SMU doesn't exist? Not sure why we need this as well.

Reply:
Thanks for responding so quickly.
Because we want the system to be workable even when SMU doesn't exist.
And whether syscon driver is ready is a different scenario.

Reply:
Ok, I think I got the point here. So, you can still enable the L2 cache with `andes_v5_l2c_enable` even when the `syscon` is not available?

Reply:
Yes.

Reply:
`syscon` is always defined in DT, are there any cases where the IP is not available? If not, any reason to not enable it?

Reply:
Thanks for your interest!

Hope the following description will make it clear:
For our hardware, we have several configurable IPs which means we can add/remove IPs for our SOCs.
As a result, we want `l2_cache.c` is not only for specific SOC. This PR is to fix the dependency between L2 cache and SMU which both are configurable IPs for our SOCs.